web_monitoring/db.py

```diff
@@ -163,6 +163,27 @@ def from_env(cls):
 Alternatively, you can instaniate Client(user, password) directly.""")
         return cls(email=email, password=password, url=url)
 
+    def get(self, url, params=None):
+        return requests.get(url,
+                            params=params,
+                            auth=self._auth,
+                            headers={'accept': 'application/json'})
+
+    def post(self, url, data, params=None):
+        return requests.post(url,
+                             data=json.dumps(data),
+                             params=params,
+                             auth=self._auth,
+                             headers={'Content-Type': 'application/json'})
+
+    def post_multi(self, url, data, params=None):
+        return requests.post(url,
+                             data='\n'.join(map(json.dumps, data)),
+                             params=params,
+                             auth=self._auth,
+                             headers={'Content-Type': 'application/x-json-stream'})
```

Comment:
Under the hood, `request.get/post/etc()` just call `requests.request('<method>', *args, **kwargs)`, so I wonder if we might not just be better off having a single method here, e.g:

```py
def request(self, method, url, **kwargs):
    kwargs.setdefault('auth', self._auth)
    kwargs.setdefault('headers', {})
    kwargs['headers'].setdefault('accept', 'application/json')

    if 'data' in kwargs:
        # Serialize data based on whether its a dict or sequence
        # and set content-type header

    return requests.request(method, url, **kwargs)
```

Or at least have each of these call into a common method like the above that just adds the commone bits (e.g. auth, the `accept` header).

That way we don’t have any repeated logic that could get out of sync (e.g. your `post` and `post_multi` methods are missing the `accept` header).

Reply:
Actually, I also forgot `requests.request` and `requests.post` can take a `json` parameter instead of `data` that will automatically get serialized as JSON and set the right header! Not sure why we weren’t using that before.

Reply:
Actually, another good option here might be to construct a [`requests.Session`](https://requests.kennethreitz.org/en/master/api/#request-sessions) instance for the `db.Client` and use that.

We do something similar for `WaybackClient` (except we have a special session class there, which we *probably* don’t need here): https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/0c91dca087f62eeb5330d9c05de6e0062f30dad3/web_monitoring/internetarchive.py#L360-L361

Reply:
Thanks for the tip on Session, I'm adopting that (and will bring in the timeout setting from #486 if combining two issues into one PR is OK).

For your comment about the json parameter, do you see any documentation indicating that it'll properly newline-separate multiple json objects and set content-type=x-json-stream? I don't see any, so I'm going to continue to use data parameter for now, but happy to drop the conditional if possible!

Reply:
Oops, didn’t realize I never replied to this! No, it won’t automatically do streaming/newline-delimited JSON, so you’d still have to do the extra work you are doing for the `POST /imports` endpoint. So keeping it the way you have it is all good — I guess it simplifies the choice of which parameter to use. :)